x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/JdbcConfigurationTests.java

```diff
@@ -113,6 +117,66 @@ public void testHttpWithSSLDisabledFromPropertyAndEnabledFromProtocol() throws E
         Exception e = expectThrows(JdbcSQLException.class, () -> ci("jdbc:es://https://test?ssl=false"));
         assertEquals("Cannot enable SSL: HTTPS protocol being used in the URL and SSL disabled in properties", e.getMessage());
     }
+    
+    public void testValidatePropertiesDefault() {
+        Exception e = expectThrows(JdbcSQLException.class, () -> ci("jdbc:es://test:9200?pagee.size=12"));
+        assertEquals("Unknown parameter [pagee.size]; did you mean [page.size]", e.getMessage());
+        
+        e = expectThrows(JdbcSQLException.class, () -> ci("jdbc:es://test:9200?foo=bar"));
+        assertEquals("Unknown parameter [foo]; did you mean [ssl]", e.getMessage());
+    }
+    
+    public void testValidateProperties() {
+        Exception e = expectThrows(JdbcSQLException.class, () -> ci("jdbc:es://test:9200?pagee.size=12&validate.properties=true"));
+        assertEquals("Unknown parameter [pagee.size]; did you mean [page.size]", e.getMessage());
+        
+        e = expectThrows(JdbcSQLException.class, () -> ci("jdbc:es://test:9200?&validate.properties=true&something=some_value"));
+        assertEquals("Unknown parameter [something]; did you mean []", e.getMessage());
+        
+        Properties properties  = new Properties();
+        properties.setProperty(PROPERTIES_VALIDATION, "true");
+        e = expectThrows(JdbcSQLException.class, () -> JdbcConfiguration.create("jdbc:es://test:9200?something=some_value", properties, 0));
```

Comment:
This seems like a test for `testValidatePropertiesDefault`, did you miss to add the `validate.properties=true`?

Reply:
@matriv the properties can be specified in a `Properties` object, as well. And in this case, it's specified like this (see above  `properties.setProperty(PROPERTIES_VALIDATION, "true");`).

Reply:
Oh, sorry, totally missed that somehow :-)